seamapi/javascript
# Return Access Codes Immediately Instead of Waiting for Action Attempt

- Add tests
- Usage example for readme (#4)
- Add test dependencies
- Generate JSON schema
- Update test config
- Rename response
- Run test workflow on push
- Check out submodules
- Use HTTP instead of SSH remote
- Update workflow
- Fix types for tests
- Increase test timeout
- skip ci: remove test
- list and get methods added for connected accounts
- fix
- Add dispatch workflow
- Run on push and PRs
- chore(release): 1.1.1 [skip ci]
- fix: update typings
- chore(release): 1.1.2 [skip ci]
- feat: add support for scheduled codes
- chore(release): 1.2.0 [skip ci]
- fix: update README
- fix: format code better
- chore(release): 1.2.1 [skip ci]
- fix: usage example
- chore(release): 1.2.2 [skip ci]
- feat: also output ESM, add CJS wrapper
- chore(release): 1.3.0 [skip ci]
- fix: update response type
- chore(release): 1.3.1 [skip ci]
- Update package.json
- chore(release): 1.3.2 [skip ci]
- await action attempts, return important seam object from endpoints
- switch to ssh cloning of submodule
- fixes based on testing
- update tests submodule
- fix access code querying, update readme
- BREAKING CHANGE: commit to fix release, fix README header
- update readme to attempt release
- last try at major release
- chore(release): 2.0.0 [skip ci]
- remove log line
- fix action attempt failures not throwing
- Add CLI framework & handlers for connected accounts, action attempts
- chore(release): 2.0.1 [skip ci]
- Refactor into separate files
- Add handler for access-codes
- Add handler for connect webview
- Add handlers for devices and locks
- Add handler for workspaces
- Tighten types
- Set up packaging
- Update README
- Split off entry point
- Build standalone binaries during release
- chore(release): 2.1.0 [skip ci]
- Import commands directly to support bundling
- Add bundle for browser
- Fix formatting
- Add shim for process.stdout.write
- Add export so both downstream Webpack and TS are happy
- Wait for command to execute, add intro to help message
- Fix doc references
- Trim command prefix
- chore(release): 2.2.0 [skip ci]
- Fix cache dir for CI builds
- feat: bump version
- chore(release): 2.3.0 [skip ci]
- fix: move typed-emitter to prod dependencies
- chore(release): 2.3.1 [skip ci]
- Update tests
- fix: clean up types
- chore(release): 2.3.2 [skip ci]
- support SEAM_API_URL env var in seam client initialization
- chore(release): 2.3.3 [skip ci]
- support for --workspace-id and --endpoint, change constructor params to options object
- revert version change
- improve browser cli options
- allow workspace id and endpoint flags in cli
- Update src/cli/browser.ts
- fix prettier
- execute command shouldn't pass undefined to allow defaults to be adopted
- chore(release): 2.4.0 [skip ci]
- fix: Introduction of `devices.update` (#19)
- chore(release): 2.4.1 [skip ci]
- Fix test dispatch?
- Use ref
- Throw custom error on malformed input
- Trigger workflow dispatch?
- Update names
- Trigger workflow dispatch?
- Trigger workflow dispatch?
- Fix workflow ref?
- chore(release): 2.5.0 [skip ci]
- fix: update type
- chore(release): 2.5.1 [skip ci]
- Add support for receiving webhooks
- chore(release): 2.6.0 [skip ci]
- Handle basic login flow
- Add 2FA support
- Rename, better error handling
- Use type package
- Fix package reference
- Rename dependency
- Add separate login command
- chore(release): 2.7.0 [skip ci]
- feat: add request ID to thrown errors
- chore(release): 2.8.0 [skip ci]
- feat: support getting a device by name
- chore(release): 3.0.0 [skip ci]
- Better formatting of validation errors
- Add validation error to message
- chore(release): 3.1.0 [skip ci]
- Remove test submodule
- Add tests
- Add test workflow
- Type fixes
- Fix broken "some examples" readme link
- Cache dependencies in workflow
- Pre-pull Seam Connect image
- Use TypeScript for AVA worker
- Parallelize seeding
- Don't need to wait for device anymore
- Bump actions/setup-node version
- feat: handle redirect panes
- fix: build script
- fix: pre-commit hook
- chore(release): 3.2.0 [skip ci]
- Add user agent
- chore(release): 3.3.0 [skip ci]
- Pin Svix fixture version and run migrations
- TEST: debug testcontainers
- Check out correct commit?
- Remove debug
- feat: Add filters to Devices List (#44)
- chore(release): 4.0.0 [skip ci]
- feat: Add accessCodes.update (#46)
- chore(release): 4.1.0 [skip ci]
- feat: Implemented webhooks endpoints (#47)
- chore(release): 4.2.0 [skip ci]
- Fix test
- feat: Add request retries (#43)
- chore(release): 4.3.0 [skip ci]
- fix: bump seamapi-types
- chore(release): 4.3.1 [skip ci]
- feat: add events list request (#48)
- chore(release): 4.4.0 [skip ci]
- fix: device typings
- chore(release): 4.4.1 [skip ci]
- feat: allow passing custom Axios options
- chore(release): 4.5.0 [skip ci]
- Use ghcr.io
- Fix org name
- Login after checkout?
- Login with bot token
- fix: Add `has_keypad` property
- chore(release): 4.6.0 [skip ci]
- fix #66: Adjust Docker container reference
- chore(release): 4.6.1 [skip ci]
- fix #62: Implement `devices.delete`
- chore(release): 4.7.0 [skip ci]
- fix #55: Implement `devices.update`
- chore(release): 4.8.0 [skip ci]
- fix #61: Implement `access_codes.get`
- chore(release): 4.9.0 [skip ci]
- fix #54: Allow getting a ConnectedAccount via email
- fix: Add `manufacturer` property to `LockProperties`
- fix: Add `manufacturer` prop to `LockProperties`
- chore(release): 4.10.0 [skip ci]
- fix: Move `UserIdentifier` above `ConnectedAccount`
- fix: Add test for getting `ConnectedAccount` by ID
- chore(release): 4.11.0 [skip ci]
- fix #53: Implement `connected_accounts.delete`
- fix #73: Implement `connect_webviews.delete`
- chore(release): 4.12.0 [skip ci]
- fix: Type error and rebuild docs
- chore(release): 4.13.0 [skip ci]
- Skiptest until seam connect issue resolved (#80)
- fixes #100
